src/dmd/glue.d

```diff
@@ -1344,6 +1256,112 @@ void FuncDeclaration_toObjFile(FuncDeclaration fd, bool multiobj)
 }
 
 
+/*******************************************
+ * Detect special functions like `main()` and do special handling for them,
+ * like special mangling, including libraries, setting the storage class, etc.
+ * `objmod` and `fd` are updated.
+ *
+ * Params:
+ *      objmod = object module
+ *      fd = function symbol
+ */
+private void specialFunctions(Obj objmod, FuncDeclaration fd)
+{
+    const(char)* libname = (global.params.symdebug)
+                            ? global.params.debuglibname
+                            : global.params.defaultlibname;
+
+    Symbol* s = fd.toSymbol();  // backend symbol corresponding to fd
+
+    // Pull in RTL startup code (but only once)
+    if (fd.isMain() && onlyOneMain(fd.loc))
+    {
+        if (global.params.isLinux || global.params.isOSX || global.params.isFreeBSD ||
+            global.params.isOpenBSD || global.params.isDragonFlyBSD || global.params.isSolaris)
+        {
+            objmod.external_def("_main");
+        }
+        else if (global.params.mscoff)
+        {
+            objmod.external_def("main");
+        }
+        else if (config.exe == EX_WIN32)
+        {
+            objmod.external_def("_main");
+            objmod.external_def("__acrtused_con");
+        }
+        if (libname)
+            objmod.includelib(libname);
+        s.Sclass = SCglobal;
+    }
+    else if (fd.isRtInit())
+    {
+        if (global.params.isLinux || global.params.isOSX || global.params.isFreeBSD ||
+            global.params.isOpenBSD || global.params.isDragonFlyBSD || global.params.isSolaris ||
+            global.params.mscoff)
```

Comment:
I know that this is just a split-out refactoring, but wouldn't `globals.useExceptions` be better here?

Reply:
Perhaps, but this PR is much easier to review as a straight cut&paste job. The minute the pasted code starts getting rewritten, too, makes it much much harder.